src/server/compositor/stream.h

```diff
@@ -45,8 +45,8 @@ class Stream : public BufferStream
     void submit_buffer(std::shared_ptr<graphics::Buffer> const& buffer) override;
     void with_most_recent_buffer_do(std::function<void(graphics::Buffer&)> const& exec) override;
     MirPixelFormat pixel_format() const override;
-    void add_observer(std::shared_ptr<scene::SurfaceObserver> const& observer) override;
-    void remove_observer(std::weak_ptr<scene::SurfaceObserver> const& observer) override;
+    void set_frame_posted_callback(
+        std::function<void(geometry::Size const&)> const& callback) override;
```

Comment:
Taking by const reference means that all the temporaries that are passed will be copied. It would be more efficient to take them by value (allowing copy elision at the call site) and std::moved() in the implementation, I can't imagine this code is critical path though.